### PR TITLE
DocumentFragment changes

### PR DESCRIPTION
**Pull request by influx6**
I needed the document Fragment API also so after checking the base documents it seems only be a simple case, would love to have this as part of the library rather than my own private fork. Thanks

Cheers

Alex

### CODE REVIEW

File: dom.go [hunk 1]
```diff
@@ -501,6 +525,26 @@ type HTMLDocument interface {
 	// TODO HTMLDocument methods
 }
 
+type htmlFragmentDocument struct {
```
Comment (dominikh) [inline dom.go]: What's this type for? It doesn't seem to be used anywhere.

File: dom.go [hunk 2]
```diff
@@ -85,6 +85,8 @@
 // TokenList will provide Set([]string) and SetString(string) methods,
 // which will be able to accomplish the same. Additionally, our
 // TokenList will provide methods to convert it to strings and slices.
+
+//TODO: removing this temporary until owner merges branch
```
Comment (dominikh) [inline dom.go]: Remove the TODO.

File: dom.go [hunk 3]
```diff
@@ -162,6 +168,14 @@ func wrapDocument(o *js.Object) Document {
 	}
 }
 
+func wrapDocumentFragment(o *js.Object) DocumentFragment {
+	switch o.Get("constructor") {
+	//TODO: do we have any other stuff we want to check
```
Comment (dominikh) [inline dom.go]: Add a space between comment marker and comment

File: dom.go [hunk 4]
```diff
@@ -470,6 +484,16 @@ type Document interface {
 	GetElementByID(id string) Element
 	QuerySelector(sel string) Element
 	QuerySelectorAll(sel string) []Element
+
+	CreateDocumentFragment() DocumentFragment
+}
+
+type DocumentFragment interface {
+	ParentNode
```
Comment (dominikh) [inline dom.go]: Swap the order of ParentNode and Node, to stay consistent with other interfaces

**Comment by dominikh** [comment]
Please squash all your commits.

The change looks okay, apart from my comments. Make the changes and I'll merge it.

**Comment by influx6** [comment]
Commits have been squashed according and fixes added. Thanks

**Comment by dominikh** [comment]
Merged as a900f4636e01f006b98f184aaba35f669bcf568f, thank you.

**Comment by influx6** [comment]
Thank you also, God bless.

**Pull request closed without merging**